Add ls.rb file to 05.ls directory

st1t opened the pull request:
- This file is an assignment submission for https://bootcamp.fjord.jp/practices/160

st1t committed:
Add ls.rb file to 05.ls directory

- This file is an assignment submission for https://bootcamp.fjord.jp/practices/160

st1t committed:
Add ls.rb file to 05.ls directory

- This file is an assignment submission for https://bootcamp.fjord.jp/practices/160

st1t commented:
# コマンド実行結果

![Screenshot from 2020-11-19 15-20-25](https://user-images.githubusercontent.com/16526902/99629136-d140e580-2a7a-11eb-8777-4e1e820e9ea9.png)

# rubocop実行結果

![Screenshot from 2020-11-19 15-21-25](https://user-images.githubusercontent.com/16526902/99629200-eae22d00-2a7a-11eb-8672-6d5051b7900c.png)

st1t commented:
すいません、なぜかファイル末尾に空白行が入ってなかったのでコミットを重ねます🙏
Rubymineで見ると124行目が存在するのですが、GitHubで見る時は改行が２行必要っぽい気がしています。。。

st1t commented:
https://docs.komagata.org/5460 の改行と、空白行をごっちゃにしてしてました、なのでこのままで問題なかったです💦
引き続きレビューをよろしくお願いします🙏

st1t commented on 05.ls/ls.rb:
```diff
@@ -0,0 +1,123 @@
+  end
+
+  def parse_options(files, option)
+    all_files = parse_all_option(files, option.has?(:all))
```
`parse_all_option`メソッドに限らず、parse_XXXX系のメソッドはオプションによって処理されたハッシュを返したりするのですが、この挙動と名前がイマイチしっくりきてないです。。。
ハッシュや配列を返すなら、`parsed_all_option_files`のほうが良かったんだろうか。。。というもやっとポイントがあります🤔

u1tnk replied:
parseというのがプログラマとしては 「フリーフォーマットの文字列を解析してデータ構造(hashとかarrayとか)にする」というイメージがあるので分かりづらかったです。
しかた無いのですが「all」という名前なのもあって「全部のオプションを解析する？いや、AppOptionクラスでやってるはずでは 🤔 」ってなってました。

私なら名前だけなら以下
```
filter_files(files, option.has?(:all))
```
処理内容的に:allのときは何もしないので
```
filter_hidden_files(files) unless option.has?(:all)
```
みたいに実行自体をしないようにします。
メソッド名も処理の内容に即した名前になると思います。
(この`filter_hidden_files`は `not_hidden_files`と同じ処理で名前だけ変更したイメージです)

参考にして他の名前も考えてみてください。

st1t replied:
ご確認ありがとうございます！
なるほど、確かに`parse`から`filter`への変更は自分の中でもしっくりきました！
メソッド名はparse_XXXX系からfilter_XXXX_filesに変更してみました。
また、返り値を受け取る変数名はfiltered_files等のフィルタが適用済みであるファイルが入ってることをわかるようにしてみました。

修正に伴う実行結果も問題ない認識です。
![Screenshot from 2020-11-24 22-20-12](https://user-images.githubusercontent.com/16526902/100099876-570dc800-2ea3-11eb-87a9-67f7f1e51109.png)

st1t committed:
Add GitHub Actions settig

- Add rubocop settings

st1t committed:
Merge pull request #2 from st1t/setup-github-actions

Add GitHub Actions settig

st1t committed:
Merge branch 'ls' of github.com:st1t/ruby-practices into ls

st1t commented:
せっかくPRベースになったので、GitHub Actionsでrubocopを実行するように設定を追加しました🙏
※今後は余力があればrubocop以外にもテストを書くこともあると思います。そのときにrubocopと合わせてtestもGitHub Actionsに仕込みたいので。
All checks have passedのリンク先に実行結果もあります。

u1tnk commented on .github/workflows/ruby.yml:
```diff
@@ -0,0 +1,32 @@
+# This workflow uses actions that are not certified by GitHub.
```
つかいこなしてますね 👍 
なるほど DevOpsの方なんですね。納得。

st1t committed:
Modify ls.rb

- Modify method names in ls.rb https://github.com/st1t/ruby-practices/pull/1#pullrequestreview-537093166
- Modify to preveent line breaks when there is no long option

st1t commented:
ご確認ありがとうございます！
コメントにもありますが、以下の視点で修正してみました！

- `parse_XXXX`系の名前の変更
    - paser -> filterに変更
- allオプションがついてない時だけfilter_hidden_filesを実行するように変更
    - parse_all_optionメソッドが不要になったため削除
- longオプションがついてない時は改行不要なので、putsからprintメソッドを使うように修正
